lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -284,538 +352,512 @@ bool PatchEntryPointMutate::isResourceNodeActive(const ResourceNode *node, bool
 }
 
 // =====================================================================================================================
-// Generates the type for the new entry-point based on already-collected info in LLVM context.
+// Generates the type for the new entry-point based on already-collected info.
+// This is what decides what SGPRs and VGPRs are passed to the shader at wave dispatch:
+//
+// * (For a GFX9+ merged shader or NGG primitive shader, the 8 system SGPRs at the start are not accounted for here.)
+// * The "user data" SGPRs, up to 32 (GFX9+ non-compute shader) or 16 (compute shader or <=GFX8). Many of the values
+//   here are pointers, but are passed as a single 32-bit register and then expanded to 64-bit in the shader code:
+//   - The "global information table", containing various descriptors such as the inter-shader rings
+//   - The "per-shader table", which is added here but appears to be unused
+//   - The streamout table if needed
+//   - Nodes from the root user data layout, including pointers to descriptor sets. In a compute shader, these
+//     must have the same layout as the root user data layout, even if that would mean leaving gaps in register
+//     usage, and are limited to 10 SGPRs s2-s11
+//   - For a compute shader, the spill table pointer if needed, which always goes into s12. (This is possibly
```

Comment:
It is a PAL limitation. it only support remap user data in graphics pipeline. I think PAL should remove this limitation in the future.